### PR TITLE
The solution of the 'expression calculator' problem

### PR DESCRIPTION
**Pull request by anahittum**

### CODE REVIEW

File: src/main/com/homeworks/week9/calculator/SimpleCalculator.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class SimpleCalculator {
+
+    public String askAndCalculate() {
+
+        String expression = new Scanner(System.in).next();
+
+        if (!expression.matches("^[0-9]+[-,*,+,/]{1}+\\d+$")) {
+            return "ERROR";
+        }
+        if (expression.contains("*")) {
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week9/calculator/SimpleCalculator.java:14]: Let's create an OperationType enum class and move calculation logic there.

File: src/main/com/homeworks/week9/calculator/SimpleCalculator.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.Scanner;
+
+public class SimpleCalculator {
+
+    public String askAndCalculate() {
+
+        String expression = new Scanner(System.in).next();
+
+        if (!expression.matches("^[0-9]+[-,*,+,/]{1}+\\d+$")) {
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week9/calculator/SimpleCalculator.java:11]: well done

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/homeworks/week9/calculator/SimpleCalculator.java (hunk 1 above)
Comment (anahittum) [reply src/main/com/homeworks/week9/calculator/SimpleCalculator.java:14]: done

File: src/main/com/homeworks/week9/calculator/OperationType.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+    },
+    DIVISION {
+        @Override
+        public String calculate(String expression) {
+            String[] parts = expression.split("/");
+            return Double.toString(Double.parseDouble(parts[0]) / Double.parseDouble(parts[1]));
+        }
+    };
+    public abstract String calculate(String expression);
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week9/calculator/OperationType.java:32]: let's change the method signature to expect 2 integer numbers

Comment (gevorgyanarman) [reply src/main/com/homeworks/week9/calculator/OperationType.java:32]: the operation enum should not be aware that it is a part of 'expression' application. it should be independent

File: src/main/com/homeworks/week9/calculator/SimpleCalculator.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            return "ERROR";
+        }
+        if (expression.contains("*")) {
+            return OperationType.MULTIPLICATION.calculate(expression);
+        }
+        if (expression.contains("-")) {
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week9/calculator/SimpleCalculator.java:17]: let's move this ifs to Operation enum class. create a method that will take a char and will return an enum value

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Pull request merged**